src/exchanges.js

```diff
@@ -527,9 +528,13 @@ Exchanges.prototype.connect = async function(options) {
   }
 
   // Find exchange prefix
+  const namespace = options.namespace || credentials.username || 'taskcluster-fake';
+  if (process.env.NODE_ENV === 'production') {
+    assert(namespace === options.projectName, 'in production, the namespace must match the projectName');
```

Comment:
Ah, neat.

Reply:
In tc-lib-pulse, we'll just enforce this, so you won't specify a separate namespace.